pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -135,8 +185,14 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 				},
 			}
 
-			err := r.client.Create(context.TODO(), mur)
+			err = r.client.Create(context.TODO(), mur)
 			if err != nil {
+				err := r.setStatusFailedToCreateMUR(instance, "Failed to create MasterUserRecord")
+				if err != nil {
+					reqLogger.Error(err, "Error creating MasterUserRecord", "UserID", instance.Spec.UserID)
```

Comment:
Here you got an error when updating the status. Not the MUR creation error.

Reply:
Fixed this (and other instances) to use a separate error variable.